### PR TITLE
Ranking page index rework and version update

### PR DESCRIPTION
**Pull request by nekkan**
This pull requests is intended to make the following updates:
* Update Kotlin and kotlinx.coroutines (1.4.10 -> 1.4.20, 1.4.1 -> 1.4.2);
* Update Gradle to use the latest performance improvements and bug fixes (6.4.2 -> 6.8);
* Make transactions 'page' variable a val instead of var using elvis operator;
* Rewrite the page retrieving in ranking commands:*
```kotlin
// old
var page = context.args.getOrNull(0)?.toLongOrNull()
    
if (page != null && !RankingGenerator.isValidRankingPage(page)) {
    context.reply(
        LorittaReply(
            locale["commands.invalidRankingPage"],
            Constants.ERROR
        )
    )
    return@executesDiscord
}

if (page != null)
    page -= 1

if (page == null)
    page = 0

// new
val pageIndex = when (val value = context.args.getOrNull(0)?.toLongOrNull()?.coerceAtLeast(0)) {
    0L, null -> 0L
    else -> {
        if (!RankingGenerator.isValidRankingPage(value)) {
            context.reply(LorittaReply(locale["commands.invalidRankingPage"], Constants.ERROR))
            return@executesDiscord
        }
        value - 1
    }
}
```

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -4,11 +4,11 @@ import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
 // val kotlinVersion by lazy { ext["kotlin-version"] as String }
 // val ktorVersion by lazy { ext["ktor-version"] as String }
 val loriVersion   = "2020-SNAPSHOT"
-val kotlinVersion = "1.4.10"
+val kotlinVersion = "1.4.20"
```
Comment (MrPowerGamerBR) [inline build.gradle.kts:7]: The latest Kotlin version is 1.4.21

**Review by MrPowerGamerBR — CHANGES_REQUESTED** [review]
Also, it is better to keep everything as TABs instead of spaces

**Comment by MrPowerGamerBR** [comment]
I personally thing that, while the new code is shorter, it is way more unreadable compared to the old one.

**Comment by nekkan** [comment]
> I personally thing that, while the new code is shorter, it is way more unreadable compared to the old one.

I don't think so, besides it is always good to maintain good practices

**Pull request closed without merging**